integration_tests/src/main/python/cache_test.py

```diff
@@ -122,7 +122,7 @@ def test_cache_reverse_order(enable_vectorized_conf):
     col1 = StructGen([['child0', byte_gen]])
     def partial_return():
         def partial_return_cache(spark):
-            return two_col_df(spark, col0, col1).select(f.col("a"), f.col("b")).cache().limit(50).select(f.col("b"), f.col("a"))
+            return two_col_df(spark, col0, col1).select(f.col("a"), f.col("b")).cache().select(f.col("b"), f.col("a"))
```

Comment:
Is the order really guaranteed without limit? Adding  `@ignore_order` looks like a more intuitive solution to me

Reply:
That will only sort the collected list AFAIK which by the time it comes back to the driver is too late, no?

Reply:
Or are you asking to add `ignore_order` in addition to the change? I can do that

Reply:
You are right about the local sort, but you can also say  `@ignore_order(local=False)`

Reply:
and you are right about removing limit, since this ends up just sorting a non-deterministic subset instead of taking the head of a sorted list.

Reply:
So, I would expect that to work but setting `local=False` still applies sort after the limit resulting in the test to fail. 

As discussed offline, I will remove the limit and add `ignore_order`